### PR TITLE
feat(unstable/publish): error when a package's module is excluded from publishing

### LINKED ISSUE #22657: Error when file in exports module graph will not be included in publish
If someone has excluded (or not included) a file from publishing that's reachable via an export, then `deno publish` should error if not already erroring.

### PR DESCRIPTION
**Pull request by dsherret**
Closes #22657

### CODE REVIEW

File: cli/tools/registry/diagnostics.rs [hunk 1]
```diff
@@ -314,57 +330,65 @@ impl Diagnostic for PublishDiagnostic {
...
         Some(Cow::Borrowed("https://jsr.io/go/unsupported-file-type"))
       }
-      PublishDiagnostic::InvalidExternalImport { .. } => {
+      InvalidExternalImport { .. } => {
         Some(Cow::Borrowed("https://jsr.io/go/invalid-external-import"))
       }
-      PublishDiagnostic::UnsupportedJsxTsx { .. } => None,
+      UnsupportedJsxTsx { .. } => None,
+      ExcludedModule { .. } => {
+        Some(Cow::Borrowed("https://jsr.io/go/excluded-module"))
```
Comment (dsherret) [inline cli/tools/registry/diagnostics.rs:390]: todo: add a page for this in the jsr registry

Comment (bartlomieju) [reply cli/tools/registry/diagnostics.rs:390]: Do you want to address this TODO before landing?

Comment (dsherret) [reply cli/tools/registry/diagnostics.rs:390]: Yes, if this url looks ok.

Comment (bartlomieju) [reply cli/tools/registry/diagnostics.rs:390]: Looks okay, but it's redirecting to the main doc page right now. I guess we'll create that page after landing this PR?

**Review by bartlomieju — APPROVED** [review]

File: cli/tools/registry/diagnostics.rs (hunk 1 above)
Comment (dsherret) [reply cli/tools/registry/diagnostics.rs:390]: I opened https://github.com/jsr-io/jsr/pull/318

**Pull request merged**